app/models/course.rb

```diff
@@ -751,6 +751,18 @@ def remove_carat_from_error_messages
     end
   end
 
+  def select_modern_language_subjects
+    subjects.select { |subject| subject.type == "ModernLanguagesSubject" }
+  end
+
+  def master_subject_nil?
+    master_subject_id.nil?
+  end
+
+  def has_any_modern_language_subject_type?
+    subjects.any? { |subject| subject.type == "ModernLanguagesSubject" }
```

Comment:
Could this just use the `modern_language_subjects` method above and check for `.present?`

Reply:
Yep, i've changed it.

Reply:
https://github.com/DFE-Digital/publish-teacher-training/pull/3113/commits/79d82a6489f1860adb88518b69236e9bb13a1b32